### PR TITLE
refactor(robot-server, app): pipette offset calibration user flow and state machine

### LINKED ISSUE #6407: Offset Calibration: Create user flow in robot server

### PR DESCRIPTION
**Pull request by b-cooper**
# Overview

Add initial pipette offset calibration infrastructure to the robot server

Closes #6407

# Changelog

- add user flow and state machine to new `sessions/calibration/pipette_offset` dir

# Review requests

- look over the code

# Risk assessment

low, currently un-called code

### CODE REVIEW

**Review by sfoster1 — CHANGES_REQUESTED** [review]
Couple little cleanup changes but it looks good!

**Comment by sfoster1** [comment]
Might want to make the title `refactor(robot-server,app):...` since there's app changes too

File: robot-server/robot_server/robot/calibration/pipette_offset/models.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    )
+
+
+class PipetteOffsetCalibrationSessionStatus(BaseModel):
+    """The current status of a pipette offset calibration session."""
+    instrument: AttachedPipette
+    currentStep: str = Field(
+        ...,
+        description="Current step of pipette offset user flow")
+    nextSteps: Optional[Dict[str, Dict[str, Dict[str, Any]]]] =\
```
[the remark this answers is not in the document]
Comment (b-cooper) [reply robot-server/robot_server/robot/calibration/pipette_offset/models.py:23]: Yeah that was copy pasta, I'll iron out the type here and from the copied location.

File: robot-server/scripts/plot_session.py [hunk 2]
```diff
@@ -14,6 +14,10 @@
     DeckCalibrationState)
 from robot_server.robot.calibration.deck.state_machine import (
     DeckCalibrationStateMachine)
+from robot_server.robot.calibration.pipette_offset.constants import (
```
Comment (sfoster1) [inline robot-server/scripts/plot_session.py:17]: Can fix the circular import here by importing `import robot_server.service.session.session_types`

File: robot-server/robot_server/robot/calibration/pipette_offset/constants.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    preparingPipette = "preparingPipette"
+    inspectingTip = "inspectingTip"
+    joggingToDeck = "joggingToDeck"
+    joggingToPointOne = "joggingToPointOne"
+    calibrationComplete = "calibrationComplete"
+    sessionExited = "sessionExited"
+    WILDCARD = STATE_WILDCARD
+
+
+JOG_TO_DECK_SLOT = '5'
```
Comment (sfoster1) [inline robot-server/robot_server/robot/calibration/pipette_offset/constants.py:17]: Could make this `Final` or `Literal['5']` but the juice may not be worth the squeeze

File: robot-server/robot_server/robot/calibration/pipette_offset/models.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    )
+
+
+class PipetteOffsetCalibrationSessionStatus(BaseModel):
+    """The current status of a pipette offset calibration session."""
+    instrument: AttachedPipette
+    currentStep: str = Field(
+        ...,
+        description="Current step of pipette offset user flow")
+    nextSteps: Optional[NextSteps] =\
```
Comment (sfoster1) [inline robot-server/robot_server/robot/calibration/pipette_offset/models.py:25]: Nice

**Review by sfoster1 — APPROVED** [review]
Looks good except for the circular import thing, and may want to wait until some other fixes are merged

**Pull request merged**